### PR TITLE
last flutter web sync: cc38319841

### PR DESCRIPTION
**Pull request by yjbanov**
This syncs Flutter Web engine to https://github.com/flutter/flutter_web/commit/cc38319841009fcbdf632d9b03aab18d28939438.

This is the last sync. Future changes will be done by submitting and reviewing pull requests in this repository.

/cc @ferhatb @hterkelsen @nturgut @mdebbar @jonahwilliams

### CODE REVIEW

File: lib/web_ui/lib/src/ui/text.dart [hunk 1]
```diff
@@ -1019,51 +1019,6 @@ enum BoxWidthStyle {
   max,
 }
 
-class LineMetrics {
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/ui/text.dart]: Seems like you'll want this?

Comment (yjbanov) [reply lib/web_ui/lib/src/ui/text.dart]: Good catch! Done.

Comment (yjbanov) [reply lib/web_ui/lib/src/ui/text.dart]: Hold on! Gotta revert a little more...

File: lib/web_ui/lib/src/engine/text/paragraph.dart [hunk 2]
```diff
@@ -290,12 +290,6 @@ class EngineParagraph implements ui.Paragraph {
     final int end = WordBreaker.nextBreakIndex(_plainText, offset);
     return <int>[start, end];
   }
-
-  @override
-  List<ui.LineMetrics> computeLineMetrics() {
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/text/paragraph.dart]: Return empty List here?

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/text/paragraph.dart]: Done. Should be good now. Once and for all!

File: lib/web_ui/lib/src/engine/text/paragraph.dart [hunk 3]
```diff
@@ -290,6 +290,12 @@ class EngineParagraph implements ui.Paragraph {
     final int end = WordBreaker.nextBreakIndex(_plainText, offset);
     return <int>[start, end];
   }
+
+  @override
+  List<ui.LineMetrics> computeLineMetrics() {
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/text/paragraph.dart]: cc @GaryQian does anything in flutter call this yet? Will returning null be safe here?

Comment (GaryQian) [reply lib/web_ui/lib/src/engine/text/paragraph.dart]: Nothing calls this yet. API just landed today/yesterday

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/text/paragraph.dart]: If you look at the overall diff for this PR you'll notice that there's no change here. IOW, returning null is exactly what we're already doing. However, I filed an issue for us to implement this properly: https://github.com/flutter/flutter/issues/39537

Comment (yjbanov) [reply lib/web_ui/lib/src/engine/text/paragraph.dart]: I updated the TODO with the issue link

**Pull request merged**